File: projects/engineering.purdue.edu/kak/scriptingwithobjects/swocode/chap4/SearchAndReplace.pl
#!/usr/bin/perl -w
  
## SearchAndReplace.pl

use strict; 

my $input_string = "one hello is like any other hello";            #(A)
$input_string =~ s/hello/armadello/g;                              #(B)
print "$input_string\n";                                           #(C)
#                        one armadello is like any other armadello

$input_string = "apples     oranges\nbananas\t   pears";           #(D)
$input_string =~ s/\s+/ /g;                                        #(E)
print "$input_string\n";                                           #(F)
#                        apples oranges bananas pears

$_ = "apples oranges, bananas; pears\n";                           #(G)
s/\b(\w+)[.,;:]\s+//g;                                             #(H)
print;                                                             #(I)
#                        apples pears 

$_ = "http://programming_with_objects.com";                        #(J)
s#//#//www.#;                                                      #(K)
print "$_\n";                                                      #(L)
#                        http://www.programming_with_objects.com

$_ = "apples oranges bananas pears";                               #(M)
s{(\ba\w+)}{\U$1}g;                                                #(N)
print;                                                             #(O)
#                        APPLES oranges bananas pears

$_ = "Ha! You are so funny!";                                      #(P)
s/^(\S+)/$1 x 3/e;                                                 #(Q)
print;                                                             #(R)
#                        Ha!Ha!Ha! You are so funny!

File: projects/demos/engineering.purdue.edu/scriptingwithobjects/swocode/chap4/SearchAndReplace.py
#!/usr/bin/env python
  
## SearchAndReplace.py

import re

input_string = "one hello is like any other hello"                  #(A)
input_string = re.sub( 'hello', 'armadello', input_string )         #(B)
print input_string                                                  #(C)
#                    one armadello is like any other armadello

input_string = "apples     oranges\nbananas\t   pears"              #(D)
input_string = re.sub( r'\s+', ' ', input_string )                  #(E)
print input_string                                                  #(F)
#                    apples oranges bananas pears

input_string = "apples oranges, bananas; pears\n"                   #(G)
output_string = re.sub( r'\b(\w+)[.,;:]\s+', r'\1 ', input_string ) #(H)
print output_string                                                 #(I)
#                    apples oranges bananas pears

input_string = "http://programming_with_objects.com"                #(J)
print re.sub( r'//', r'//www.', input_string )                      #(K)
#                    http://www.programming_with_objects.com

input_string = "apples oranges bananas pears"                       #(L)
print re.sub( r'(\ba\w+)', r'\1'.upper(), input_string )            #(M)
#                    apples oranges bananas pears

print re.sub( r'(\ba\w+)', r'hello'.upper(), input_string )         #(N)
#                    HELLO oranges bananas pears
